server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -512,6 +518,24 @@ public Indexer(String indexName,
         this.expressions = ctxForRefs.expressions();
     }
 
+    /**
+     * @param addedColumns has columns collected by {@link #collectSchemaUpdates(IndexItem) and with assigned OID0-s
+     */
+    public void updateTargets(List<Reference> addedColumns) {
+        // Store by ident for fast replacement.
+        Map<ColumnIdent, Reference> addedColumnsByIdent = new HashMap<>();
+        for (Reference ref: addedColumns) {
+            addedColumnsByIdent.put(ref.column(), ref);
+        }
```

Comment:
Initially `AddColumnResponse` returned `Map<ColumnIdent, Reference> . 
Since map can be built from a reference and needed only for faster "intersection" computation, made response return list and building map here.